processing/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -777,14 +802,17 @@ protected void compute()
    */
   static class QueuePusher<E> implements ForkJoinPool.ManagedBlocker
   {
+    static final long BLOCK_TIMEOUT = TimeUnit.MILLISECONDS.convert(500, TimeUnit.NANOSECONDS);
```

Comment:
something is off here: this is `500 nanosec` in `millis` ; which is `0` :D

I guess this wanted to be:
```
TimeUnit.MILLISECONDS.toNanos(500)
```

Reply:
yes, i wrote this backwards, good catch